x-pack/test/functional_ccs/apps/lens/smokescreen_ccs.ts

```diff
@@ -0,0 +1,796 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { range } from 'lodash';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const PageObjects = getPageObjects(['visualize', 'lens', 'common', 'header']);
+  const find = getService('find');
+  const listingTable = getService('listingTable');
+  const testSubjects = getService('testSubjects');
+  const elasticChart = getService('elasticChart');
+  const filterBar = getService('filterBar');
+  const retry = getService('retry');
+
+  describe('lens smokescreen tests', () => {
```

Comment:
This file is almost exact copy of https://github.com/elastic/kibana/blob/af38b9db6cbd15490caa1ebffcbeaf5c6336eb9c/x-pack/test/functional_ccs/apps/lens/smokescreen_ccs.ts

<img width="1430" alt="Screenshot 2022-03-23 at 10 05 29" src="https://user-images.githubusercontent.com/10977896/159664526-c852fa07-32e1-46b8-b8cb-7cb46338cf75.png">

if this test suite is basically 99% same as original one, can we consider load original file and modify it to use the required data view for verifcation?


```
loadTestFile(require.resolve('./../../../functional/apps/lens/smokescreen_ccs.ts'));
```

If lens functionality will be updated, we don't want to update the same tests in multiple files, do we?

Reply:
Agreed, it would be cool to keep them together as close as possible.

Reply:
@dmlemeshko Originally, the intention was to replace test files with their CCS counterparts but after meeting with @stacey-gammon, @spenceralger , @LeeDr  it was decided that a small subset of tests that exercise a good amount of an app's functionality would be used and configured in this way. We are not going to be redoing entire suites of tests. We are taking one test from each app that can use the CCS feature to ensure that CCS works properly when using the given app's functionality. The duplication of files was discussed and it was decided upon that this would be accepted.

Reply:
This issue is part of a larger issue, https://github.com/elastic/kibana/issues/126392

Reply:
> The duplication of files was discussed and it was decided upon that this would be accepted.

As the VisEditors team will have to own and maintain these tests from now on, I would prefer to deduplicate as much as possible to reduce the maintenance burden to a minimum. I'm not saying the test should only run for CCS, I was thinking about a file which exports a function which can be parameterized for both cases.

So there's

`smokescreen_template.ts`:

```ts
export function createTests(indexPattern: string) {
 //...
   it('should allow to change index pattern', async () => {
      await PageObjects.lens.switchFirstLayerIndexPattern(indexPattern);
      expect(await PageObjects.lens.getFirstLayerIndexPattern()).to.equal(indexPattern);
    });
  // ...
}
```

and then there's `smokescreen.ts`:
```ts
describe('regular smokescreen', () => {
  // some setup
  createTests('log*');
});
```

and then there's `smokescreen_ccs.ts`:
```ts
describe('ccs smokescreen', () => {
  // some setup
  createTests('ftr-remote:log*');
});
```

I don't really see a downside of this approach, with the upside that future changes to this test (which will happen for sure, it was changed 13 times in the last year) only have to be done once instead of copy/pasted to multiple files.

Happy to discuss if there are other opinions.

Reply:
Went a little bit of a slightly different route but still wound up parameterizing the test file and keeping things to a single test file rather than duplicating. Let me know what you think. @flash1293 @dmlemeshko @wayneseymour @LeeDr